model/src/main/kotlin/licenses/LicenseInfoResolver.kt

```diff
@@ -98,23 +102,7 @@ class LicenseInfoResolver(
                 }.keys
 
                 licenseInfo.declaredLicenseInfo.authors.takeIf { it.isNotEmpty() && addAuthorsToCopyrights }?.also {
-                    locations += ResolvedLicenseLocation(
-                        provenance = UnknownProvenance,
-                        location = UNDEFINED_TEXT_LOCATION,
-                        appliedCuration = null,
-                        matchingPathExcludes = emptyList(),
-                        copyrights = it.mapTo(mutableSetOf()) { author ->
-                            val statement = "Copyright (C) $author".takeUnless {
-                                author.contains("Copyright", ignoreCase = true)
-                            } ?: author
-
-                            ResolvedCopyrightFinding(
-                                statement = statement,
-                                location = UNDEFINED_TEXT_LOCATION,
-                                matchingPathExcludes = emptyList()
-                            )
-                        }
-                    )
+                    locations += getAuthorsAsCopyrightTextLocation(it)
```

Comment:
As the change is quite complex to understand (at least for me), I'd appreciate doing more fine-granular commit steps:

1. Extract the existing code to a function and announce the upcoming reuse in the commit message. I believe the function name should not include the word "Text". I prose "resolveCopyrightFromAuthors" or similar.
2. Add a disabled test that fails. That allows to easily checkout that commit and enable the test to quickly convince yourself that there is an issue, and what it is about.
3. Apply the fix together with enabling the test.

Reply:
Okay, I split it up in 3 separate commits to make more transparent what are pure refactorings, and which parts actually change the license/copyright resolution process logic.